### ISSUE TITLE
Peer Review - Mac

### ISSUE THREAD

**Issue by AustinSeto** [OpenLiberty/draft-guide-jax-rs-reactive-client#45]
Functionality
---
- [x] Check that the guide has working instructions and sample code
- [ ] If any URL visits, try them on different browsers: Firefox, Chrome, Safari
- [x] If any URL visits, try `curl` command where applicable 

Formatting & Presentation
---

`README.adoc` checks:
- [x] Ensure that command substitutions use the `$(cmd)` format and not ``` `cmd` ```
- [x] Ensure that the console output of the integrated tests do not include the `[INFO]` tab
- [x] Check the quality and presentation of guide according to [Structure and Style Guideline](https://github.com/OpenLiberty/draft-guides-template/wiki/Guidelines-for-Structure-and-Styles)
- [x] Check that the pom.xml, server.xml, etc files are clean
    - [x] Remove any dependencies and features that are not required
    - [x] Ensure that the parent pom.xml files do not contain any dependencies, and all dependencies are specified in the children pom.xml files
- [x] Examples with the right outcomes are provided
- [x] Example with the wrong outcomes, if any, are provided

`pom.xml` checks (if files are present):
- [x] Check that the file follows the template found here: https://github.com/OpenLiberty/draft-guides-template/blob/master/finish/pom.xml
- [x] Check that the following dependencies are included in all of the `pom.xml` files:
```
<!-- Provided dependencies -->
<dependency>
  <groupId>jakarta.platform</groupId>
  <artifactId>jakarta.jakartaee-api</artifactId>
  <version>8.0.0</version>
  <scope>provided</scope>
</dependency>

<dependency>
  <groupId>org.eclipse.microprofile</groupId
  <artifactId>microprofile</artifactId>
  <version>3.3</version>
  <type>pom</type>
  <scope>provided</scope>
</dependency>
```
- [x]  Check that the versions of plugins are directly provided and not fed in by a file property.
  - Not formatted like so:
```
    <properties>
        <!-- Plugins -->
        <version.liberty-maven-plugin>3.2</version.liberty-maven-plugin>
    </properties>

    <!-- Liberty plugin -->
    <plugin>
        <groupId>io.openliberty.tools</groupId>
        <artifactId>liberty-maven-plugin</artifactId>
        <version>${version.liberty-maven-plugin}</version>
    </plugin>
```
  - Should be formatted like so:
```
    <!-- Liberty plugin -->
    <plugin>
        <groupId>io.openliberty.tools</groupId>
        <artifactId>liberty-maven-plugin</artifactId>
        <version>3.2</version>
    </plugin>
```
- [x] Check that there are 4 spaces per indent for proper formatting
- [x] Ensure that the guide is using the latest version of the `liberty-maven-plugin` or `liberty-gradle-plugin` where applicable

Overall checks:
- [x] Check the consistency of guide with the template and other guides
- [ ] Check the quality of code according to the best coding practices
- [x] Check that all licensing statements are properly stated in all files, with the correct year (Should be present in all Java files + the `index.html`)
- [x] Check that the directories are properly structured
- [ ] Check that some of these `page-tags` are used in a guide: `MicroProfile, Maven, Docker, Kubernetes, Gradle, Java EE, Security, Cloud`. Only these tags are visible on the website. Latest list [here](https://github.com/OpenLiberty/openliberty.io/blob/master/src/main/content/_includes/visible-tags.liquid). 
- [x] Check the `attribution` statement is accurate for the guide
- [x] Verify the `travisTest.sh` script, if any, is accurate and consistent with other guides

---

Guide’s contributor’s (if available, otherwise peer tester’s) responsibility: 
- [ ] Additional tests where applicable: 
    - [ ] Define test coverage and review with team (including guide contributor, if available)
    - [ ] Define detail test cases 
    - [ ] Consider corner cases targeting the specific guide
    - [ ] Consider corner cases UI tests
    - [ ] Consider testing URL on all browsers, ie, FF, Chrome, Safari
    - [ ] Consider testing the `curl` command for URL visits
- [ ] Consider building with both Maven and Gradle build tools 
- [ ] Testing with different IDEs, ie, Atom, Eclipse (Optional: VS.code, IntelliJ, Microclimate)
- [ ] Run Acrolinx Checker on draft (above 70 score approximately)
- [x] Consider SEO title and description for the guides 
- [ ] Ensure automated test is enable, set up with Travis CI, and able to schedule tests 
- [ ] Run `diff -r start/ finish/` and there's no differences
- [x] Ensure that the automation tests are able to run when PR is created

Peer Tester’s responsibility: 
- [ ] Check the appearance of the guide on test site for the following items: 
    - [x] Table of contents 
    - [x] Headings
    - [x] Paragraphs
    - [x] code snippets
    - [x] outputs
    - [x] links
    - [ ] hotspots 
- [x] Test the guide end-to-end with working instruction and sample code
- [x] Perform all the defined test cases

**Comment by AustinSeto** [comment]
# README

- [ ] Change uses of "`CompletionStage<T>` and `Observable<T>` to just `CompletionStage` or `Observable` (or vice versa) for consistency

## "What is Reactive Programming" section

- [ ] Remove "among others" from line "The usage of reactive streams brings with it tools, among others, to handle data flow, backpressure, thread safety and error propagation."

## "Using the default reactive JAX-RS client" section

- [ ] May be worth mentioning that the `thenAcceptAsync` method runs the provided function when the computation is complete

## "Updating the client to use third-party libraries" section

- [ ] Suggest this rewording (changes in bold) "**The** `rxjava` **dependency** allows the application to use the RxJava API.  **The** `jersey-rx-client-rxjava` and `jersey-rx-client-rxjava2` **dependencies** provide"
- [ ] Suggest rewording "In cases like these, JAX-RS can deal with this issue using `backpressure` and `Flowables`" to "using RxJava’s `Flowable` with `backpressure`" to match wording of link

## "Updating the client to use the reactive model" section

- [ ] Hotspot for `holder` does not highlight the `updateHighest` and `updateLowest` method calls

## "Testing the query microservice" section

- [ ] Description of test case doesn't need list format if there's only one test case

**Comment by AustinSeto** [comment]
# Code

- [ ] `QueryResource.java` uses a wildcard import, should avoid using wildcards
- [ ] Guide says to create `QueryServiceIT.java` but the file already exists in the `start` folder

**Comment by TiagoF99** [comment]
> Hotspot for holder does not highlight the updateHighest and updateLowest method calls

I dont think we should highlight those two lines because the context of the line where I highlight `Holder` is that I only wanna show the actual object. So i show the object class and the object being invoked. I think showing the other two methods calls would be out of place

> Have instruction to add various dependencies instead of just including them in the given pom.xml
I will investigate this alongside gilbert
